src/interpreter/FunctionRegistry.ts

```diff
@@ -30,12 +49,8 @@ export class FunctionRegistry {
   }
 
   public static registerFunction(functionId: string, plugin: FunctionPluginDefinition, translations?: FunctionTranslationsPackage): void {
-    const entry = plugin.implementedFunctions[functionId]
-    if (entry !== undefined) {
-      this.loadPluginFunction(plugin, functionId, this.plugins)
-    } else {
-      throw FunctionPluginValidationError.functionNotDeclaredInPlugin(functionId, plugin.name)
-    }
+    validateAndReturnMetadataFromName(functionId, plugin)
+    this.loadPluginFunction(plugin, functionId, this.plugins)
```

Comment:
Is `validateAndReturnMetadataFromName` necessary here? We will call it again in `loadPluginFunction` -> `loadFunctionUnprotected` first line. Calling those checks twice looks like an unnecessary overhead